goaviatrix/cloudn_registration.go

```diff
@@ -0,0 +1,65 @@
+package goaviatrix
+
+import (
+	"context"
+)
+
+type CloudnRegistration struct {
+	CID               string `form:"CID"`
+	Action            string `form:"action"`
+	ControllerAddress string `form:"controller_ip_or_fqdn"`
+	Username          string `form:"username"`
+	Password          string `form:"password"`
+	Name              string `form:"gateway_name"`
+	PrependAsPath     []string
+}
+
+func (c *Client) CreateCloudnRegistration(ctx context.Context, cloudnRegistration *CloudnRegistration) error {
```

Comment:
Can we add a comment to tell us that this method should be called on a CloudN only, not a regular controller.